File: CS/CustomMVCHelpers/Helpers/DevExpressHelpers.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web.Mvc;

namespace CustomMVCHelpers.Helpers
{
    public static class DevExpressHelpers
    {
        public static MvcHtmlString GridHelper(this HtmlHelper helper, string name, Type ModelType, string ControllerName)
        {
           
            var PropertiesArray = ModelType.GetProperties();
            var g = new GridViewSettings()
            { Name = name,
                KeyFieldName = PropertiesArray.Where(pr => pr.GetCustomAttributes(typeof(KeyAttribute), true) != null).FirstOrDefault().Name,
                CallbackRouteValues = new
            { Controller = ControllerName, Action = name + "Partial" } };
            // for demonstration purposes
            g.CommandColumn.Visible = false;          
            g.SettingsEditing.UpdateRowRouteValues = new
            { Controller = ControllerName, Action = name + "UpdatePartial" };
            g.SettingsEditing.DeleteRowRouteValues = new
            { Controller = ControllerName, Action = name + "DeletePartial" };
            g.SettingsEditing.AddNewRowRouteValues = new
            { Controller = ControllerName, Action = name + "InsertPartial" };
            g.Settings.ShowFilterRow = true;
            g.Settings.ShowGroupPanel = true;
            foreach (var item in PropertiesArray)
            {
                var c =  g.Columns.Add(item.Name);
                if (c.FieldName == g.KeyFieldName)
                {
                    c.EditFormSettings.Visible =  DevExpress.Utils.DefaultBoolean.False;
                }             
            }
            new GridViewExtension(g, helper.ViewContext).Bind(helper.ViewData.Model).Render();          
            return MvcHtmlString.Empty;
        }
       
    }
}

File: VB/CustomMVCHelpers/Helpers/DevExpressHelpers.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Web.Mvc
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Web.Mvc

Namespace CustomMVCHelpers.Helpers
	Public Module DevExpressHelpers             
        <System.Runtime.CompilerServices.Extension> _
        Public Function GridHelper(ByVal helper As HtmlHelper, ByVal name As String, ByVal ModelType As Type, ByVal ControllerName As String) As MvcHtmlString

            Dim PropertiesArray = ModelType.GetProperties()
            Dim g = New GridViewSettings() With {.Name = name, .KeyFieldName = PropertiesArray.Where(Function(pr) pr.GetCustomAttributes(GetType(KeyAttribute), True) IsNot Nothing).FirstOrDefault().Name, .CallbackRouteValues = New With {Key .Controller = ControllerName, Key .Action = name & "Partial"}}
            g.CommandColumn.Visible = False
            ' for demonstration purposes
            g.SettingsEditing.UpdateRowRouteValues = New With {Key .Controller = ControllerName, Key .Action = name & "UpdatePartial"}
            g.SettingsEditing.DeleteRowRouteValues = New With {Key .Controller = ControllerName, Key .Action = name & "DeletePartial"}
            g.SettingsEditing.AddNewRowRouteValues = New With {Key .Controller = ControllerName, Key .Action = name & "InsertPartial"}
            g.Settings.ShowFilterRow = True
            g.Settings.ShowGroupPanel = True
            For Each item In PropertiesArray
                Dim c = g.Columns.Add(item.Name)
                If c.FieldName = g.KeyFieldName Then
                    c.EditFormSettings.Visible = DevExpress.Utils.DefaultBoolean.False
                End If
            Next item
            CType(New GridViewExtension(g, helper.ViewContext), GridViewExtension).Bind(helper.ViewData.Model).Render()

            Return MvcHtmlString.Empty
        End Function
		
		
	End Module
End Namespace
